linalg/ode.cpp

```diff
@@ -9,12 +9,167 @@
 // terms of the BSD-3 license. We welcome feedback and contributions, see file
 // CONTRIBUTING.md for details.
 
+#include "../general/communication.hpp"
 #include "operator.hpp"
 #include "ode.hpp"
 
 namespace mfem
 {
 
+std::string ODESolver::ExplicitTypes = "\n\tExplicit solver: \n\t"
+                                       "        RK      :  1 - Forward Euler, 2 - RK2(0.5), 3 - RK3 SSP, 4 - RK4,\n\t"
+                                       "        AB      : 11 - AB1, 12 - AB2, 13 - AB3, 14 - AB4, 15 - AB5\n";
+
+std::string ODESolver::ImplicitTypes  = "\n\tImplicit solver: \n\t"
+                                        "        (L-Stab): 21 - Backward Euler, 22 - SDIRK23(2), 23 - SDIRK33,\n\t"
+                                        "        (A-Stab): 32 - Implicit Midpoint, 33 - SDIRK23, 34 - SDIRK34,\n\t"
+                                        "        GA      : 40 -- 50  - Generalized-alpha,\n\t"
+                                        "        AM      : 51 - AM1, 52 - AM2, 53 - AM3, 54 - AM4\n";
+
+std::string ODESolver::Types = ODESolver::ExplicitTypes +
+                               ODESolver::ImplicitTypes;
+
+/// Return a ODESolver pointer based on an type
+/// Caller gets ownership of the object and is responsible for its deletion
+ODESolver* ODESolver::Select(int ode_solver_type)
+{
+   if (ode_solver_type< 20)
+   {
+      return SelectExplicit(ode_solver_type);
+   }
+   else
+   {
+      return SelectImplicit(ode_solver_type);
+   }
+}
+
+/// Return a ODESolver pointer based on an type
+/// Caller gets ownership of the object and is responsible for its deletion
+ODESolver* ODESolver::SelectExplicit(int ode_solver_type)
+{
+   ODESolver*  ode_solver = NULL;
+   switch (ode_solver_type)
+   {
+      // Explicit RK methods
+      case 1: ode_solver = new ForwardEulerSolver; break;
+      case 2: ode_solver = new RK2Solver(0.5); break; // midpoint method
+      case 3: ode_solver = new RK3SSPSolver; break;
+      case 4: ode_solver = new RK4Solver; break;
+
+      // Explicit AB methods
+      case 11: ode_solver = new AB1Solver; break;
+      case 12: ode_solver = new AB2Solver; break;
+      case 13: ode_solver = new AB3Solver; break;
+      case 14: ode_solver = new AB4Solver; break;
+      case 15: ode_solver = new AB5Solver; break;
+
+      default:
+         mfem::out<<" ODE solver type: " << ode_solver_type << '\n';
+         mfem_error("Unknown ODE solver type");
```

Comment:
Better to use `MFEM_ABORT` instead

Reply:
Done.

Reply:
One more minor suggestion (here and in the other instances in this file), with MFEM_ABORT we can combine the two lines:

```c++
MFEM_ABORT("Unknown ODE solver type " << ode_solver_type);
```

Reply:
Will make the change.